### PR TITLE
Whitelist

### PR DESCRIPTION
**Pull request by Adaobi-Chuks**

### CODE REVIEW

**Review by king-Alex-d-great — CHANGES_REQUESTED** [review]

File: lib/handlers/whitelist-handler.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const Joi = require("joi");
+const { whitelistModel } = require("../models/whitelist-model");
+const { BAD_REQUEST, OK } = require("../utils/status-codes");
+const { isTemporaryEmail } = require("../utils/helper-utils");
+
+exports.addToWhitelist = async (request, response, next) => {
+    const { email } = request.body;
+
+    const whitelistSchema = Joi.object({email: Joi.string().email().required()});
```
[the remark this answers is not in the document]
Comment (Adaobi-Chuks) [reply lib/handlers/whitelist-handler.js:7]: Thank you for your feedback, I now use `value.email` to find by ID and check for temporary email, also used it in creating a new whitelist model.

File: lib/handlers/whitelist-handler.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        let errorMessage = [];
+        error.details.forEach(detail => {
+            errorMessage.push(detail.message);
+        })
+        return response.status(BAD_REQUEST).send({ message: errorMessage })
+    }
+    if(await whitelistModel.findById({_id: email})) {
+        return response.status(BAD_REQUEST).send({ message: "Email already exist" })
+    }
+    if (isTemporaryEmail(email)) {
```
[the remark this answers is not in the document]
Comment (Adaobi-Chuks) [reply lib/handlers/whitelist-handler.js]: Done

File: lib/middlewares/isWhitelisted.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const { whitelistModel } = require("../models/whitelist-model");
+const { BAD_REQUEST, OK } = require("../utils/status-codes");
+
+exports.isWhitelisted = async (request, response, next) => {
+    const { _id } = request.user;
+    console.log(request.user)
```
[the remark this answers is not in the document]
Comment (Adaobi-Chuks) [reply lib/middlewares/isWhitelisted.js]: Let's pretend like this didn't happen😁

File: lib/routes/v1/auth-routes.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 const { Router } = require("express");
 const { authenticate, login, loginWithMagicLink } = require("../../handlers/auth-handler");
+const { isWhitelisted } = require("../../middlewares/isWhitelisted");
 
 module.exports = Router()
 .get("/", authenticate)
 .get("/login", login)
-.post("/magic-login", loginWithMagicLink);
+.post("/magic-login", isWhitelisted, loginWithMagicLink);
```
[the remark this answers is not in the document]
Comment (Adaobi-Chuks) [reply lib/routes/v1/auth-routes.js]: So I just went ahead to validate the user in the handler instead, hope this is better.

**Review by king-Alex-d-great — APPROVED** [review]

**Pull request merged**